fixtures_test.go

```diff
@@ -1,403 +1,131 @@
-          "detail_type": {
-            "code": "CONFIRMATION_ONLY"
-          }
-        },
-        {
-          "summary": "Yolo tours are designed for travellers between the ages of 18-39.",
-          "description": "Yolo tours are designed for those who thirst for adventure and want to take on the planet in a fast, raw and affordable way. These tours make the most of the time available, hitting the highlights by day, and mixing it up at night. These trips often feature public transportation, and basic accommodations and have been designed for those on a budget. Please note that Yolo style tours are designed for travellers aged 18-39.",
-          "detail_type": {
-            "code": "ANY"
-          }
-        }
-      ]
-    },
-    {
-      "type": "CHECKIN",
-      "name": "Passport Expiry Date",
-      "code": "PASSPORT_EXPIRY_DATE",
-      "message": "Passport expiry date must be submitted for this product. The 'expiry_date' field (in the 'passport' data) must be set in the customer resource.",
-      "flags": [],
-      "details": []
-    },
-    {
-      "type": "CHECKIN",
-      "name": "Passport Number",
-      "code": "PASSPORT_NUMBER",
-      "message": "Passport number must be submitted for this product. The 'number' field (in the 'passport' data) must be set in the customer resource.",
-      "flags": [],
-      "details": []
-    },
-    {
-      "type": "INFORMATIONAL",
-      "name": "Customer Information",
-      "code": "CUSTOMER_INFORMATION",
-      "message": "Information that must be passed along to the customer but is not actionable.",
-      "flags": [],
-      "details": []
-    }
-  ],
-  "components": {
-    "href": "https://rest.gadventures.com/departures/733048/departure_components"
-  }
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"testing"
+)
+
+func loadTestData(filename string) (string, error) {
+	file, err := os.Open(fmt.Sprintf("./testdata/%s", filename))
+	if err != nil {
+		return "", fmt.Errorf("failed to load file %s - %s", filename, err.Error())
+	}
+	defer file.Close()
+	blob, err := io.ReadAll(file)
+	if err != nil {
+		return "", fmt.Errorf("failed to read file %s - %s", filename, err.Error())
+	}
+	return string(blob), nil
+}
+
+// hold the data in memory
+var departureBlob, errorBlob string
+
+func setup() {
+	var err error
+	departureBlob, err = loadTestData("departure.json")
+	if err != nil {
+		panic(err)
+	}
+	errorBlob, err = loadTestData("gapi_error.json")
+	if err != nil {
+		panic(err)
+	}
+}
+
+func TestMain(m *testing.M) {
+	setup()
+	os.Exit(m.Run())
+}
```

Comment:
```suggestion
func loadTestData(t *testing.T, name string) string {
	t.Helper()
	data, err := os.ReadFile(fmt.Sprintf("testdata/%s", name))
	if err != nil {
		t.Fatal(err)
	}
	return string(data)
}
```

and then the tests can call `data := loadTestData(t, "fname.json")` directly. By using `os.ReadFile`, we can consolidate the opening and the reading of the file into a single step. By using `t.Helper()`, we can make it easier to use in tests, while still getting useful error messages.

Actually, since the string data is always wrapped into a `bytes.NewBufferString` in the tests, `loadTestData` could return `[]bytes` directly, and then the tests could use ` bytes.NewReader` instead.